FIX Add CSV download link to stock at date table

bradley-jarvis opened the pull request:
This update adds a csv download link to the stock at date list

Commit:
FIX Add CSV download link to stock at date table

Commit:
Update array values to strings for joining

Commit:
Fix array and string formatting

Commit:
Fix space after comma

eldy commented on htdocs/product/stock/.htaccess:
```diff
@@ -0,0 +1,7 @@
+RewriteEngine On
```
Adding  files with specific setup to an external software and environment variable is not portable.
Can you remove this file.

eldy commented on htdocs/product/stock/stockatdate.php:
```diff
@@ -353,164 +359,187 @@
+	if ($productid > 0) {
+		$param .= '&productid='.(int) $productid;
+	}
+	if (GETPOST('dateday', 'int') > 0) {
```
Why replacing GETPOSTINT int GETPOST(... , 'int') ?

eldy commented on htdocs/product/stock/stockatdate.php:
```diff
@@ -353,164 +359,187 @@
+
+	print '<div class="div-table-responsive">'; // You can use div-table-responsive-no-min if you don't need reserved height for your table
+	if ($num) {
+		print "<p><a href=\"stockatdate.csv?sortfield=$sortfield&sortorder=$sortorder&type=$type&mode=$mode".
```
A dynamic page must end with .php, using another extension will generate troubles when using proxy or cache server.
You can add the parameter  &output=file  to provide the information you want a download.

Commit:
FIX stock at date csv download

Removed .htaccess and requirement for modrewrite
Added output url parameter which takes csv as argument
updated GETPOST(,'int') calls to GETPOSTINT()

Commit:
FIX stock at date csv download

Removed .htaccess and requirement for modrewrite
Added output url parameter which takes csv as argument
updated GETPOST(,'int') calls to GETPOSTINT()

Commit:
Merge branch 'fix-add-stockatdate-csv-download' of github.com:bradley-jarvis/dolibarr into fix-add-stockatdate-csv-download

bradley-jarvis commented:
That now makes sense why you don't use mod rewrite anywhere else, updated code as requested. The GETPOSTINT/GETPOST revert was an accident and also now fixed.

eldy commented on htdocs/product/stock/stockatdate.php:
```diff
@@ -353,164 +359,187 @@
+	if (GETPOSTINT('dateday') > 0) {
+		$param .= '&dateday='.GETPOSTINT('dateday');
+	}
+	if (GETPOSTINT('datemonth', 'int') > 0) {
```
With GETPOSTINT, there is no need for the second parameter 'int'.

Commit:
FIX removed extra 'int' parameter from GETPOSTINT()

eldy commented on htdocs/product/stock/stockatdate.php:
```diff
@@ -582,135 +611,178 @@
-		if (!getDolGlobalString('PRODUIT_MULTIPRICES')) {
-			print '<td class="right">'.price(price2num($totalsellingprice, 'MT')).'</td>';
+		if ($mode == 'future') {
+			print '<td class="right">'.$totalcurrentstock.'</td>';
```
The $totalcurrentstock is a result of operation between float value so result is a float with a non exact decimal part. For this reason me must use price(price2num($totalcurrentstock, 'MS'))
price2num to trunc the decimal part that is not relevant and introduced by float number calculation. Then price to format the number into a number using the language of the user.

bradley-jarvis replied:
This would be the same for the totalvirtualstock value as well?

eldy replied:
> This would be the same for the totalvirtualstock value as well? what is the difference beween the 'MS' and 'MT' value?

Yes same.MT is when we need to trunc decimal part for an amount, MS is used for a stock.

bradley-jarvis replied:
That makes sense, I have now wrapped all the stock values after initially missing some. Hopefully this sorts it out

Commit:
Fix wrapped stock values in price(price2num($stock,'MS'))

Commit:
Fix error in php code

Commit:
Fix missed wrapping some stock levels tin price(...)

bradley-jarvis commented:
I have made the changes requested, so this should be all good to go now.

eldy committed:
Update stockatdate.php
